src/main/java/uk/gov/justice/digital/delius/transformers/OffenderTransformer.java

```diff
@@ -316,4 +305,147 @@ private Human humanOf(String forename, String forename2, String surname) {
+
+                // Determine whether the staff role is current for this offender and in what capacity
+                boolean isCurrentRo = (pom.getResponsibleOfficer() != null && pom.getResponsibleOfficer().getEndDate() == null);
+                boolean isCurrentPom = (pom.getActiveFlag().intValue() == 1 && pom.getEndDate() == null);
+
+                responsibleOfficers.add (
+                        uk.gov.justice.digital.delius.data.api.ResponsibleOfficer
+                                .builder()
+                                .nomsNumber(offender.getNomsNumber())
+                                .responsibleOfficerId((isCurrentRo ? pom.getResponsibleOfficer().getResponsibleOfficerId() : null))
+                                .prisonOffenderManagerId(pom.getPrisonOffenderManagerId())
+                                .staffCode(Optional.ofNullable(pom.getStaff()).map(staff -> staff.getOfficerCode()).orElse(null))
+                                .surname(Optional.ofNullable(pom.getStaff()).map(staff -> staff.getSurname()).orElse(null))
+                                .forenames(combinedMiddleNamesOf(pom.getStaff().getForename(), pom.getStaff().getForname2()).stream().collect(Collectors.joining(" ")))
+                                .providerTeamCode(Optional.ofNullable(pom.getTeam()).map(team -> team.getCode()).orElse(null))
+                                .providerTeamDescription(Optional.ofNullable(pom.getTeam()).map(team -> team.getDescription()).orElse(null))
+                                .lduCode(Optional.ofNullable(pom.getTeam()).map(team -> team.getLocalDeliveryUnit().getCode()).orElse(null))
+                                .lduDescription(Optional.ofNullable(pom.getTeam()).map(team -> team.getLocalDeliveryUnit().getDescription()).orElse(null))
+                                .probationAreaCode(Optional.ofNullable(pom.getProbationArea()).map(pa -> pa.getCode()).orElse(null))
+                                .probationAreaDescription(Optional.ofNullable(pom.getProbationArea()).map(pa -> pa.getDescription()).orElse(null))
+                                .isCurrentRo(isCurrentRo)
+                                .isCurrentOm(false)
+                                .isCurrentPom(isCurrentPom)
+                                .omStartDate(null)
+                                .omEndDate(null)
+                                .build()
+                );
+            }
+        }
+
+        if (current && !responsibleOfficers.isEmpty()) {
+            // Filter the list to only those officers who hold a currently active role for this offender
+            return responsibleOfficers
+                    .stream()
+                    .filter(ro -> (ro.isCurrentOm() || ro.isCurrentPom() || ro.isCurrentRo()))
+                    .collect(Collectors.toList());
+        }
+
+        return responsibleOfficers;
```

Comment:
Feels quite a chunky method this

Reply:
While looking at the streaming I will also see if I can break this down - I agree, it is chunky with two blocks of quite similar code.